plugins/code-coverage/src/components/CoverageHistoryChart/CoverageHistoryChart.tsx

```diff
@@ -69,6 +69,11 @@ const getTrendIcon = (trend: number, classes: ClassNameMap) => {
   }
 };
 
+// convert timestamp to human friendly form
+function formatDateToHuman(timeStamp: string | number) {
+  return new Date(timeStamp).toUTCString();
```

Comment:
The preferred way here would be to use Luxon as the date library to help in formatting as per this ADR: https://backstage.io/docs/architecture-decisions/adrs-adr010

I believe it might be worth discussing the ADR a bit further and decide the correct date format to be displayed on the UI. currently there are a range of approaches in use in the codebase so a standardized approach would be needed here. I would be proponent of using Luxon's `toLocaleString` with `DATETIME_SHORT` or `DATETIME_SHORT_WITH_SECONDS` for these items where the whole datetime is good to be visible. Additionally for other items like latest updates/runs I would propose using an 'x minutes/hours ago' style with a title showing the actual timestamp as locale string with `DATETIME_SHORT_WITH_SECONDS` as the format.

Here are the API lines for those two formats:

    /**
     * {@link DateTime.toLocaleString} format like '10/14/1983, 9:30 AM'. Only 12-hour if the locale is.
     */
    static get DATETIME_SHORT(): typeof Formats.DATETIME_SHORT;

    /**
     * {@link DateTime.toLocaleString} format like '10/14/1983, 9:30:33 AM'. Only 12-hour if the locale is.
     */
    static get DATETIME_SHORT_WITH_SECONDS(): typeof Formats.DATETIME_SHORT_WITH_SECONDS;

Reply:
It would probably make sense to make a helper component (and corresponding helper function) that we export for general use. Something like

```tsx
<Timestamp value={DateTime.fromX...} variant="short" />

type TimestampVariant = 'short' |...;

function stringifyTimestamp(value: DateTime, variant: TimestampVariant): string {
  ...
}
```

Reply:
This could go well with a future extension where we let users select whether to show local or UTC time. For e.g. log display, you often want to show UTC no matter where you are. And humanizing durations could also be standardized through the same type of mechanism of a component and a function

Reply:
Hmm seem like a can-o-worms got unintentionally opened here. While the ADR says

> All core packages and plugins within Backstage should use Luxon for any date manipulation or formatting that cannot be easily accomplished with the native JavaScript Date object.

The use of `toUTCString()` seemed to fit the second part there. I'm game for changing the code to use Luxon and what ever desired format. I do find having the date in UTC helpful since not everyone is in the same time zone and UTC provides a common zone for everyone. 

Outside of this PR what would be nice is to have a per-user setting to allow them to show time stamps in their chosen time zone.

Reply:
I should also note that I think having the TZ identifier in the final output in-order to limit the number of "what time zone is this?" questions. Looking at the example Luxon code above, it's not clear as to which timezone the date it referring to, which leaves one to guessing.

Reply:
There is actually already an ADR here for this https://backstage.io/docs/architecture-decisions/adrs-adr012

Reply:
Although it's not referenced in the sidebar, I will fix that up now.

Reply:
@benjdlambert thanks for the link. I'll update the PR to match the guideline.

Reply:
Updated PR 

UI dates now look like
<img width="1477" alt="Screen Shot 2021-11-08 at 9 20 36 AM" src="https://user-images.githubusercontent.com/23618053/140790012-1e7b80bc-947d-4a2f-b300-50f936bed486.png">